patch-package
--- a/node_modules/d3-celestial/celestial.js
+++ b/node_modules/d3-celestial/celestial.js
@@ -3922,5 +3922,5 @@ function d3_eventDispatch(target) {
 }
 
 })();
-this.Celestial = Celestial;
+if (typeof define === "function" && define.amd) this.Celestial = Celestial, define(Celestial); else if (typeof module === "object" && module.exports) module.exports = Celestial; else this.Celestial = Celestial;
 })();
\ No newline at end of file
