patch-package
--- a/node_modules/@types/vis/index.d.ts
+++ b/node_modules/@types/vis/index.d.ts
@@ -20,6 +20,13 @@ export type MomentConstructor2 =
   (inp?: MomentInput, format?: MomentFormatSpecification, language?: string, strict?: boolean) => Moment;
 
 export type MomentConstructor = MomentConstructor1 | MomentConstructor2;
+export class Graph3d {
+    constructor(container: HTMLElement,
+                items: any,
+                options?: any);
+
+    setCameraPosition(pos);
+}
 
 export type IdType = string | number;
 export type SubgroupType = IdType;
